### DISCUSSION TITLE
Can I rewrite the response URL in `+page.server.ts` without using `redirect`? Using `history.replaceState()` outside `onMount()` doesn't work

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#9025] [Q&A]
I could do this in the client `onMount()` and `history.replaceState()` or in `+page.server.ts` with `redirect()`. I find both options ok-ish but not optimal.

So, is there a way to rewrite the request URL when responding without using aforementioned methods?

**Comment by 205g0** [comment]
In this context another question: Only a `history.replaceState(null, '', '?someParam=0')` in `onMount()` will add the param to the url after load. If I do this outside latter, it doesn't work or the param just shows up a millisecond and vanishes again.

I need this to work without `onMount()` but I guess it's not possible or do I miss something?

**Reply by teemingc** [reply]
It's not possible to access `history` outside the browser context. Therefore, it's only possible in an `onMount` or `if browser` block.

**Comment by teemingc** [comment]
What's your reason for not wanting to use redirect to append the search param to the URL?

**Reply by 205g0** [reply]
The redirect adds another roundtrip but yeah, it would be the proper solution, also locating respective code to the right side. I want basically to rewrite searchParams, so it's not a full change of scenes.

**Reply by teemingc** [reply]
You could also use `goto("?someParam=0", { replace: true })` which is the svelte equivalent for `history.replaceState`.
https://kit.svelte.dev/docs/modules#$app-navigation-goto
